### PR TITLE
Consolidate exhaustiveness-related tests

### PR DESCRIPTION
**Pull request by Nadrieril**
I hunted for tests that only exercised the match exhaustiveness algorithm and regrouped them. I also improved integer-range tests since I had found them lacking while hacking around.
The interest is mainly so that one can pass `--test-args patterns` and catch most relevant tests.

r? @varkor
@rustbot modify labels: +A-exhaustiveness-checking

### CODE REVIEW

File: src/test/ui/pattern/usefulness/integer-ranges/reachability.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        10 => {},
+        11..=23 => {},
+        19..30 => {},
+        5..25 => {}, //~ ERROR unreachable pattern
+        _ => {},
+    }
+    match 0usize {
+        0..10 => {},
+        10..20 => {},
+        5..15 => {}, // FIXME: should be unreachable
```
Comment (varkor) [inline src/test/ui/pattern/usefulness/integer-ranges/reachability.rs:75]: Huh, I'm surprised our tests didn't cover this before. Good catch (and below).

Comment (varkor) [reply src/test/ui/pattern/usefulness/integer-ranges/reachability.rs:75]: We don't have an issue for this already, do we?

**Review by varkor — APPROVED** [review]
It's always good to see more comprehensive tests (especially ones that catch some missing cases)! Thanks!

File: src/test/ui/pattern/usefulness/integer-ranges/reachability.rs (hunk 1 above)
Comment (Nadrieril) [reply src/test/ui/pattern/usefulness/integer-ranges/reachability.rs:75]: We don't, but I've got a PR ready to fix it

Comment (Nadrieril) [reply src/test/ui/pattern/usefulness/integer-ranges/reachability.rs:75]: I don't know how to solve the other FIXME so I guess we could raise an issue. But it's quite tiny, I'm not sure it's worth it

Comment (varkor) [reply src/test/ui/pattern/usefulness/integer-ranges/reachability.rs:75]: It's worth opening an issue, just so we can keep track of it.

Comment (Nadrieril) [reply src/test/ui/pattern/usefulness/integer-ranges/reachability.rs:75]: Done https://github.com/rust-lang/rust/issues/79307

**Pull request merged**